test/e2e/specs/editor/various/rich-text.spec.js

```diff
@@ -0,0 +1,826 @@
+/**
+ * WordPress dependencies
+ */
+const { test, expect } = require( '@wordpress/e2e-test-utils-playwright' );
+
+test.describe( 'RichText', () => {
+	test.beforeEach( async ( { admin } ) => {
+		await admin.createNewPost();
+	} );
+
+	test( 'should handle change in tag name gracefully', async ( {
+		page,
+		editor,
+	} ) => {
+		// Regression test: The heading block changes the tag name of its
+		// RichText element. Historically this has been prone to breakage,
+		// because the Editable component prevents rerenders, so React cannot
+		// update the element by itself.
+		//
+		// See: https://github.com/WordPress/gutenberg/issues/3091
+		await editor.insertBlock( { name: 'core/heading' } );
+		await editor.clickBlockToolbarButton( 'Change level' );
+		await page.locator( 'button[aria-label="Heading 3"]' ).click();
+
+		expect( await editor.getBlocks() ).toMatchObject( [
+			{
+				name: 'core/heading',
+				attributes: { level: 3 },
+			},
+		] );
+	} );
+
+	test( 'should apply formatting with primary shortcut', async ( {
+		page,
+		editor,
+		pageUtils,
+	} ) => {
+		await editor.canvas.click( 'role=button[name="Add default block"i]' );
+		await page.keyboard.type( 'test' );
+		await pageUtils.pressKeys( 'primary+a' );
+		await pageUtils.pressKeys( 'primary+b' );
+
+		expect( await editor.getBlocks() ).toMatchObject( [
+			{
+				name: 'core/paragraph',
+				attributes: { content: '<strong>test</strong>' },
+			},
+		] );
+	} );
+
+	test( 'should apply formatting when selection is collapsed', async ( {
+		page,
+		editor,
+		pageUtils,
+	} ) => {
+		await editor.canvas.click( 'role=button[name="Add default block"i]' );
+		await page.keyboard.type( 'Some ' );
+		await pageUtils.pressKeys( 'primary+b' );
+		await page.keyboard.type( 'bold' );
+		await pageUtils.pressKeys( 'primary+b' );
+		await page.keyboard.type( '.' );
+
+		expect( await editor.getBlocks() ).toMatchObject( [
+			{
+				name: 'core/paragraph',
+				attributes: { content: 'Some <strong>bold</strong>.' },
+			},
+		] );
+	} );
+
+	test( 'should apply multiple formats when selection is collapse', async ( {
```

Comment:
```suggestion
	test( 'should apply multiple formats when selection is collapsed', async ( {
```
I was finding by titles to jump back and forth and compare old tests to new and so didn’t find this one right away.